_posts/2024-03-19-credential-stores-on-client-side.adoc

```diff
@@ -0,0 +1,155 @@
+---
+layout: post
+title: 'Using Credential Stores on for WildFly Client'
+date: 2023-03-19
+tags: credential-store elytron-tool wildfly-config ejb
+synopsis: How to set up a credential store using the elytron-tool and use it for the client configuration on an application deployed to WildFly. 
+author: prarthonapaul
+---
+
+:toc: macro
+:toc-title:
+
+WildFly allows the use of credential stores to keep alias for sensitive information, such as, passwords for external services. Credential stores can be used to store different credentials under aliases and use credential-reference to specify them in server configuration. As a result, the credential is no longer visible in clear text. However, there are additional uses for credential store as well. This guides explains how to create credential stores using the elytron-tool and use it for an application deployed to WildFly. 
+
+toc::[]
+
+== Prerequisite
+To follow along with this guide you will need:
+
+* about 10 minutes
+* WildFly with credential-store support  
+* Elytron-tool for WIldFly server
+
+== Access the Example Application
+For this guide we will be using the `ejb-security` example application. Let us first clone the project and navigate to it: 
+
+```
+git clone https://github.com/wildfly-security-incubator/elytron-examples
+cd ejb-security
+```
+
+== About WildFly Client Configuration
+EJBs, also known as Enterprise JavaBeans are a collection of specifications that are used for building java applications and offer a set og APIs for developing and running secured applications. When invoking EJBs using the WildFly server, we need to configure the WildFly client to specify relevant information about authentication to secure the application. This can be done using a file named wildfly-config.xml located inside the `src/main/resources` folder of the application. 
```

Comment:
Just a minor, should this sentence be:

When invoking EJBs using the WildFly server, we need to configure the WildFly client to specify relevant information about authentication to the secured the application. 

? Since we are not securing the application with this client config file, but we are supplying the required client authentication that should be used to authenticate the client to the secured app